File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.py
def solution(s):
    nums = {'one': '1', 'two': '2', 'three': '3', 'four': '4', 'five': '5', 'six': '6', 'seven': '7', 'eight': '8', 'nine': '9', 'zero': '0'}

    answer = s
    for k, v in nums.items():
        answer = answer.replace(k, v)
    return int(answer)

File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.java
import java.util.*;
class Solution {
    public int solution(String s) {
        
        Map<Integer, String> alphabets = new HashMap<>();
        alphabets.put(0, "zero");
        alphabets.put(1, "one");
        alphabets.put(2, "two");
        alphabets.put(3, "three");
        alphabets.put(4, "four");
        alphabets.put(5, "five");
        alphabets.put(6, "six");
        alphabets.put(7, "seven");
        alphabets.put(8, "eight");
        alphabets.put(9, "nine");
        
        StringBuilder current = new StringBuilder();
        StringBuilder result = new StringBuilder();
        
        for (int i = 0; i < s.length(); i++) {
            char character = s.charAt(i);
            
            if (Character.isDigit(character)) {
                result.append(character);
                continue;
            }
            
            current.append(character);
            for (Integer key : alphabets.keySet()) {
                if (alphabets.get(key).equals(current.toString())) {
                    result.append(key);
                    current.setLength(0);
                    break;
                }
            }
        }
        return Integer.parseInt(result.toString());
    }
}